src/util/settings/shortcuts.cc

```diff
@@ -0,0 +1,473 @@
+  }
+}
+
+QVariant ShortcutsModel::headerData(int section, Qt::Orientation orientation,
+                                    int role) const {
+  if (orientation != Qt::Orientation::Horizontal || role != Qt::DisplayRole) {
+    return QVariant();
+  }
+
+  switch(section) {
+    case COLUMN_INDEX_NAME:
+      return "Name";
+    case COLUMN_INDEX_SHORTCUTS:
+      return "Shortcuts";
+    default:
+      return QVariant();
+  }
+
+}
+
+ShortcutsModel* ShortcutsModel::getShortcutsModel() {
+  static ShortcutsModel* ptr = new ShortcutsModel();
+  return ptr;
+}
+
+QAction* ShortcutsModel::createQAction(ShortcutType type, QObject* parent, Qt::ShortcutContext context) {
+  return type_to_shortcut_[type]->createQAction(parent, context);
+}
+
+QAction* ShortcutsModel::createQAction(ShortcutType type, QObject* parent, QIcon icon, Qt::ShortcutContext context) {
+  return type_to_shortcut_[type]->createQAction(parent, icon, context);
+}
+
+void ShortcutsModel::addShortcut(ShortcutType type, QKeySequence& shortcut) {
+  if (shortcut.isEmpty()) {
+    return;
+  }
+  auto item = type_to_shortcut_.value(type);
+  if (!item) {
+    return;
+  }
+  if (item->addShortcut(shortcut)) {
+    if (sequence_to_shortcut_.contains(shortcut) && !sequence_to_shortcut_[shortcut].contains(item)) {
+      if (sequence_to_shortcut_[shortcut].size() > 0) {
+        if (sequence_to_shortcut_[shortcut].size() == 1) {
+          sequence_to_shortcut_[shortcut][0]->setConflict(true);
+        }
+        item->setConflict(true);
+      }
+      sequence_to_shortcut_[shortcut].append(item);
+    } else {
+      sequence_to_shortcut_[shortcut].append(item);
+    }
+    auto index = indexFromItem(item);
+    emit dataChanged(index, index);
+  }
+}
+
+void ShortcutsModel::removeShortcut(ShortcutType type, const QKeySequence& shortcut) {
+  if (shortcut.isEmpty()) {
+    return;
+  }
+  auto item = type_to_shortcut_.value(type);
+  if (!item) {
+    return;
+  }
+  if (item->deleteShortcut(shortcut)) {
+    sequence_to_shortcut_[shortcut].removeAll(item);
+    if (sequence_to_shortcut_[shortcut].size() == 1 && !checkIfConflicts(sequence_to_shortcut_[shortcut][0])) {
+      sequence_to_shortcut_[shortcut][0]->setConflict(false);
+    }
+    if (!checkIfConflicts(item)) {
+      item->setConflict(false);
+    }
+    auto index = indexFromItem(item);
+    emit dataChanged(index, index);
+  }
+}
+
+ShortcutsModel::ShortcutsModel() : root_(new ShortcutsItem()) {
+  auto global = addCategory(tr("Global"), root_);
+  addShortcutType(EXIT_APPLICATION, global, tr("E&xit"), tr("Exit"));
+  addShortcutType(OPEN_FILE, global, tr("&Open..."), tr("Open file"));
+//  Uncomment once supported
+//  addShortcutType(NEW_FILE, global, tr("&New..."), tr("New File"));
+  addShortcutType(SHOW_DATABASE, global, tr("Show database view"));
+  addShortcutType(SHOW_LOG, global, tr("Show log view"));
+  addShortcutType(SHOW_OPTIONS, global, tr("&Options"), tr("Show the Dialog to select applications options"));
```

Comment:
In theory they could be different strings (though I used the same :P) - will do.